src/main/app/forms/validation/validators/isValidPostcode.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments
+} from 'class-validator'
+
+import * as validator from 'validator'
+
+@ValidatorConstraint()
+export class IsValidPostcodeConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments): boolean {
+    if (value === undefined || value === null) {
+      return true
+    }
+    const UK_POSTCODE_REGEX = /^([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z]))))[0-9][A-Za-z]{2})$/
+    const normalised = value.toString().replace(/\s/g,'')
+    return validator.matches(normalised, new RegExp(UK_POSTCODE_REGEX))
```

Comment:
I think you can remove `new RegExp(` as `UK_POSTCODE_REGEX` is an regex already.